packages/components/Menu/src/MenuItem/MenuItem.types.ts

```diff
@@ -60,17 +60,20 @@ export interface MenuItemProps extends Omit<IWithPressableOptions<IViewProps>, '
   persistOnClick?: boolean;
 }
 
-export interface MenuItemState extends IPressableHooks<MenuItemProps & React.ComponentPropsWithRef<any>> {
-  hasCheckmarks?: boolean;
+export interface MenuItemState {
+  props: MenuItemProps & React.ComponentPropsWithRef<any>;
+  state: PressableState & {
+    hasCheckmarks?: boolean;
 
-  /**
-   * If the menu item is a trigger for a submenu
-   */
-  hasSubmenu?: boolean;
+    /**
+     * If the menu item is a trigger for a submenu
+     */
+    hasSubmenu?: boolean;
+  };
 }
```

Comment:
FYI, made this change to match other components. A better name might be "MenuItemInfo" but 🤷🏾‍♂️